File: back-end/student-api/src/main/java/org/p2kyoo/student/model/Student.java
package org.p2kyoo.student.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.p2kyoo.student.model.enums.Gender;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    private String phoneNumber;

    private String address;

    private int age;

    @CreationTimestamp
    private LocalDate created;

    @UpdateTimestamp
    private LocalDate updated;

    @Enumerated(EnumType.STRING)
    private Gender gender;

}

File: front-end/src/app/model/student.ts
export class Student {
  id: number;
  fullName: string;
  gender: string;
  phoneNumber: string;
  address: string;
  age: number;
  createdAt: Date;
  updatedAt: Date;


  constructor(id: number, fullName: string, gender: string, phoneNumber: string, address: string, age: number, createdAt: Date, updatedAt: Date) {
    this.id = id;
    this.fullName = fullName;
    this.gender = gender;
    this.phoneNumber = phoneNumber;
    this.address = address;
    this.age = age;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }
}
